tests/optim/test_oss.py

```diff
@@ -470,6 +470,11 @@ def closure():
     _ = optimizer.step(closure=closure)
     check_same_models_across_ranks(model, dist.group.WORLD, params_should_be_equal=True, check_broadcast_buffers=False)
 
+    # Check that if the model is moved to cpu, the optimizer consolidation still works
+    model.cpu()
```

Comment:
without the fix, this unit test does fail with the same error that the user mentioned